src/components/Header/SpacesDropdownMenuItem.tsx

```diff
@@ -0,0 +1,23 @@
+'use client';
+
+import { Space } from '@gitbook/api';
+import { useSelectedLayoutSegment } from 'next/navigation';
+
+import { DropdownMenuItem } from './Dropdown';
+
+export function SpacesDropdownMenuItem(props: { variantSpace: Space; currentSpace: Space }) {
+    const { variantSpace, currentSpace } = props;
+    const currentPathname = useSelectedLayoutSegment() ?? '';
+    const targetUrl = new URL(variantSpace.urls.published ?? variantSpace.urls.app);
+    targetUrl.pathname += `/${currentPathname}`;
```

Comment:
<img width="341" alt="Screenshot 2024-07-10 at 13 58 36" src="https://github.com/GitbookIO/gitbook/assets/701222/4b162bfb-656d-46ae-bb34-68f0a6194e13">

Will it create double slashes?

Reply:
It relies on the URL implementation which I think is a standard so we should be fine.. BUT because this code could run on the server, client etc. I think it's best to be safe here.

Reply:
But you're appending to the pathname:

```
const targetUrl = new URL(variantSpace.urls.published ?? variantSpace.urls.app); // targetUrl.pathname is "/"
targetUrl.pathname += `/${currentPathname}`; // targetUrl.path is "/" + "/hello-world"
```

Shouldn't it be:

```suggestion
    targetUrl.pathname = `/${currentPathname}`;
```

Reply:
No we need to append it because `variantSpace.urls.published ?? variantSpace.urls.app` can already contain a pathname that we need to preserve, e.g
```
https://gitbook-open-e2e-sites.gitbook.io/api-multi-versions/v/2.0
```

or for share links:

```
https://gitbook-open-e2e-sites.gitbook.io/api-multi-versions-share-links/bRfQbzwsK8rbN1GRxx7K/v/2.0
```

btw if I fed to new URL the value returned by `variantSpace.urls.published ?? variantSpace.urls.app` I'm not getting doubleslashes 
![Screenshot 2024-07-10 at 15 22 48](https://github.com/GitbookIO/gitbook/assets/1927278/82d26331-009f-4800-b149-4161aafec555)

I wondering if it's because, in this case urls.published is returned with a non root path unlike your example where you didn't provide one. 

Going to sanitize the pathname anyway but we definitely need to prepend it instead of overwritting it.

Reply:
Makes sense 👍  - yes I think a quick sanitization will help as we can never be 100% sure of the URL implementation we're dealing with. Anyways PR is approved so feel free to merge when you're happy with it.